src/Runner.Common/RunnerService.cs

```diff
@@ -68,6 +68,19 @@ protected async Task<VssConnection> EstablishVssConnection(Uri serverUrl, VssCre
             throw new InvalidOperationException(nameof(EstablishVssConnection));
         }
 
+        protected async Task RetryRequest(Func<Task> func,
```

Comment:
This is added so that this can be called for Tasks that don't have any return type. To achieve that (since we can't do `Task<void>`), it's a general pattern to use something like `Unit` that represents absent value. I added `Unit` class and used it here.